Source/JavaScriptCore/jit/RegisterAtOffset.h

```diff
@@ -28,32 +28,40 @@
 #if ENABLE(ASSEMBLER)
 
 #include "Reg.h"
+#include <cstddef>
 #include <wtf/PrintStream.h>
 
 namespace JSC {
 
 class RegisterAtOffset {
 public:
-    RegisterAtOffset()
-        : m_offset(0)
-    {
-    }
+    RegisterAtOffset() = default;
     
-    RegisterAtOffset(Reg reg, ptrdiff_t offset)
-        : m_reg(reg)
-        , m_offset(offset)
+    RegisterAtOffset(Reg reg, ptrdiff_t offset, Width width)
+        : m_regIndex(reg.index())
+        , m_width(width == Width128 ? true : false)
+        , m_offsetBits((offset >> 2) & 0xFFFFFFFFFFFFFF)
     {
+        ASSERT(!(offset & 0b11));
+        ASSERT(width == Width64 || (width == Width128 && Options::useWebAssemblySIMD()));
+        ASSERT(reg.index() < (1 << 6));
+        ASSERT(Reg::last().index() < (1 << 6));
+        ASSERT(this->reg() == reg);
+        RELEASE_ASSERT(this->offset() == offset);
```

Comment:
Why is it `RELEASE_ASSERT`? I don't think it needs to be that (ASSERT is enough).